crates/voicevox_core_c_api/src/lib.rs

```diff
@@ -98,6 +98,29 @@ pub unsafe extern "C" fn voicevox_open_jtalk_rc_new(
     })())
 }
 
+/// OpenJtalkにユーザー辞書を読み込ませる。
+/// この関数を呼び出した後にユーザー辞書を変更した場合、再度この関数を呼び出す必要がある。
+/// @param [in] open_jtalk 参照カウントで管理されたOpenJtalk
+/// @param [in] user_dict ユーザー辞書
+///
+/// # Safety
+/// @open_jtalk 有効な :OpenJtalkRc のポインタであること
+/// @user_dict 有効な :VoicevoxUserDict のポインタであること
+#[no_mangle]
+pub extern "C" fn voicevox_open_jtalk_rc_load_user_dict(
+    open_jtalk: &mut OpenJtalkRc,
+    user_dict: &VoicevoxUserDict,
+) -> VoicevoxResultCode {
+    into_result_code_with_error((|| {
+        let user_dict = user_dict.to_owned();
+        {
+            let dict = user_dict.dict.as_ref().lock().expect("lock failed");
+            open_jtalk.open_jtalk.load_user_dict(&dict)?;
+        }
```

Comment:
こことかの処理の末尾で`VoicevoxUserDict`のロックを`{}`で囲っているのって、強い理由があったりしますか?

```suggestion
        let dict = user_dict.dict.as_ref().lock().expect("lock failed");
        open_jtalk.open_jtalk.load_user_dict(&dict)?;
```

Reply:
あまりないですね、自分はMutex周り書く時はロックを最小限にする癖（？）があるのでそうしたはずです

Reply:
まあどちらでもいいと思うんですけど、しいて言うなら @sevenc-nanashi さんの方が書き方がぶれないかなと思いました！
後ろの方というのが人によってぶれるからです。
（まあ本当にどっちでもいいと思います！強いて言うなら、くらい！）